client/src/main/java/io/pravega/client/stream/ReaderGroup.java

```diff
@@ -143,6 +144,16 @@
      * @return Map of streams that this group is reading from to the corresponding cuts.
      */
     Map<Stream, StreamCut> getStreamCuts();
+
+    /**
+     * Fetch the latest StreamCut after co-ordinating with all the readers using {@link io.pravega.client.state.StateSynchronizer}
+     *
+     * @param backgroundExecutor A thread pool that will be used to poll if the positions from all the readers have been
+     *                           fetched.
+     * @return A future to a Map of streams that this group is reading from to the correspoding cuts.
+     */
+    @Beta
+    CompletableFuture<Map<Stream, StreamCut>> getCurrentStreamCut(ScheduledExecutorService backgroundExecutor);
```

Comment:
Can I have multiple cuts being coordinated concurrently? If so, how do I differentiate each one of them?

Reply:
Yes, checkpoints solve this. Let's piggy back on that same code path, and just make this a special type of checkpoint.